metricbeat/module/mongodb/dbstats/dbstats.go

```diff
@@ -88,22 +83,22 @@ func (m *MetricSet) Fetch(reporter mb.ReporterV2) {
 		err := db.Run("dbStats", &result)
 		if err != nil {
 			err = errors.Wrapf(err, "Failed to retrieve stats for db %s", dbName)
-			logger.Error(err)
+			reporter.Error(err)
+			m.Logger().Error(err)
 			continue
 		}
 		data, _ := schema.Apply(result)
-		if reported := reporter.Event(mb.Event{MetricSetFields: data}); !reported {
-			logger.Debug("error reporting event")
-			return
+		reported := reporter.Event(mb.Event{MetricSetFields: data})
+		if !reported {
```

Comment:
We could log on debug level that the metricset was stopped during reporting, but I think that can already be seen in other places, so +1 on this change.

Nit: I preferred the previous 1 line statement ;-)